From 10f60e4e2e477ae3fd6c26c31d4753cbd412d84b Mon Sep 17 00:00:00 2001
From: nagachika <nagachika@b2dd03c8-39d4-4d8f-98ff-823fe69b080e>
Date: Sun, 2 Mar 2014 16:33:36 +0000
Subject: [PATCH 5/5] merge revision(s) r45225,r45240: [Backport #9578]

	* ext/readline/readline.c (Init_readline): Use rl_hook_func_t instead
	  of Function to support readline-6.3.  (rl_hook_func_t is available
	  since readline-4.2.)
	  Reported by Dmitry Medvinsky.  [ruby-core:61141] [Bug #9578]

	* ext/readline/extconf.rb (rl_hook_func_t): define as Function for
	  very old readline versions.  [ruby-core:61209] [Bug #9578]


git-svn-id: svn+ssh://ci.ruby-lang.org/ruby/branches/ruby_2_0_0@45252 b2dd03c8-39d4-4d8f-98ff-823fe69b080e
---
 ChangeLog               | 12 ++++++++++++
 ext/readline/extconf.rb |  8 ++++++++
 ext/readline/readline.c |  2 +-
 version.h               |  2 +-
 4 files changed, 22 insertions(+), 2 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 8a22461..f789fc8 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,15 @@
+Mon Mar  3 01:32:14 2014  Nobuyoshi Nakada  <nobu@ruby-lang.org>
+
+	* ext/readline/extconf.rb (rl_hook_func_t): define as Function for
+	  very old readline versions.  [ruby-core:61209] [Bug #9578]
+
+Mon Mar  3 01:32:14 2014  Tanaka Akira  <akr@fsij.org>
+
+	* ext/readline/readline.c (Init_readline): Use rl_hook_func_t instead
+	  of Function to support readline-6.3.  (rl_hook_func_t is available
+	  since readline-4.2.)
+	  Reported by Dmitry Medvinsky.  [ruby-core:61141] [Bug #9578]
+
 Mon Mar  3 01:25:28 2014  Nobuyoshi Nakada  <nobu@ruby-lang.org>
 
 	* eval.c (setup_exception): preserve errinfo across calling #to_s
diff --git a/ext/readline/extconf.rb b/ext/readline/extconf.rb
index 4920137..74e34d4 100644
--- a/ext/readline/extconf.rb
+++ b/ext/readline/extconf.rb
@@ -19,6 +19,10 @@ def readline.have_func(func)
   return super(func, headers)
 end
 
+def readline.have_type(type)
+  return super(type, headers)
+end
+
 dir_config('curses')
 dir_config('ncurses')
 dir_config('termcap')
@@ -93,4 +97,8 @@ readline.have_func("remove_history")
 readline.have_func("clear_history")
 readline.have_func("rl_redisplay")
 readline.have_func("rl_insert_text")
+unless readline.have_type("rl_hook_func_t")
+  $DEFS << "-Drl_hook_func_t=Function"
+end
+
 create_makefile("readline")
diff --git a/ext/readline/readline.c b/ext/readline/readline.c
index 502e73e..fe2ad78 100644
--- a/ext/readline/readline.c
+++ b/ext/readline/readline.c
@@ -1883,7 +1883,7 @@ Init_readline()
 
     rl_attempted_completion_function = readline_attempted_completion_function;
 #if defined(HAVE_RL_PRE_INPUT_HOOK)
-    rl_pre_input_hook = (Function *)readline_pre_input_hook;
+    rl_pre_input_hook = (rl_hook_func_t *)readline_pre_input_hook;
 #endif
 #ifdef HAVE_RL_CATCH_SIGNALS
     rl_catch_signals = 0;
diff --git a/version.h b/version.h
index 5bda012..7ec3f66 100644
--- a/version.h
+++ b/version.h
@@ -1,6 +1,6 @@
 #define RUBY_VERSION "2.0.0"
 #define RUBY_RELEASE_DATE "2014-03-03"
-#define RUBY_PATCHLEVEL 455
+#define RUBY_PATCHLEVEL 456
 
 #define RUBY_RELEASE_YEAR 2014
 #define RUBY_RELEASE_MONTH 3
-- 
1.9.0

